test/reduce.js

```diff
@@ -287,15 +286,31 @@ module.exports = function reduce_test(testcase, minify_options, reduce_options)
                     CHANGED = true;
                     return List.skip;
                 }
+
+                // remove this node unless its the sole element of a (transient) sequence
+                if (!(parent instanceof U.AST_Sequence) || parent.expressions.length > 1) {
+                    node.start._permute++;
+                    CHANGED = true;
+                    return List.skip;
+                }
             }
 
-            // replace or remove this node depending on whether it's in a list
+            // replace this node
             var newNode = new U.parse(REPLACEMENTS[node.start._permute % REPLACEMENTS.length | 0], {
                 expression: true,
             });
             newNode.start._permute = ++node.start._permute;
             CHANGED = true;
-            return in_list ? List.skip : newNode;
+            return newNode;
+        }, function(node, in_list) {
+            if (node instanceof U.AST_Sequence) {
+                if (node.expressions.length == 1) return node.expressions[0];
```

Comment:
This sequence of length 1 only exists because it is ill formed? If so, a comment is in order here.

Reply:
I've commented on it over here:
https://github.com/mishoo/UglifyJS2/pull/3710/files#diff-28c42a2a2ff0ec81eda66ad80750958cR290

Reply:
I did see that. Was thinking of just:
```
                 // fix single element transient sequence
                 if (node.expressions.length == 1) return node.expressions[0];
```

Reply:
Added comments to all three patching cases.

Reply:
👍